docs/api-docs/in-depth-guides/rail-integration-guide.mdx

```diff
@@ -1,100 +1,154 @@
+
+We provide a set of milestones/events that let you track the status of your containers as they move through the rail system.  You can be notified by webhook whenever these events occur.
+
+We also provide a set of attributes that let you know the current status of your container at any given time, as well as useful information such as ETA, pickup facility, and availability information.
 
-Terminal49 tracks various milestones and data attributes that are essential for effective rail container management. The key events and attributes include:
+### Rail Milestones (aka rail events)
 
+{/* TODO: create rail-carriers page or section */}
+There are several core milestones that occur on most rail journeys.  Some rail carriers do not share all events, {/* (see [Rail Carriers](#rail-carriers) for more details), */} but in general these are the key events for a container.
 
 ```mermaid
 graph LR
+A[Rail Loaded] --> B[Rail Departed]
+B --> C[Rail Arrived]
+C --> D[Rail Unloaded]
+D --> E[Full Out]
+E --> F[Empty Return]
+```
+
+However, not all shipments have a simple journey.  You may want to track events such as when a train passes through a town, when the container switches trains or rail carriers at an interchange, and when the status of your container at the terminal changes.
+
+This is a more complex diagram that shows the various events that can occur in a container's rail journey.
+
+```mermaid
+graph LR
+    Y[From ocean terminal or carrier] --> A
+    Z[Rail Interchange Received] --> B
     A[Rail Loaded] --> B[Rail Departed]
+    B --> L[Train Passing]
     B --> C[Rail Arrived]
+    L --> L
+    L --> C
     C --> D[Rail Unloaded]
-    D --> E[Full Out]
+    C --> X[Rail Interchange Delivered]
+    X --> Z
+    D --> G[Available for Pickup]
+    D --> H[Not Available]
+    H -- Holds and Fees Updated --> H
+    G --> H
+    H --> G
+    G --> E[Full Out]
     E --> F[Empty Return]
-  ```
+```
 
-```mermaid
-timeline
-    title Container Milestones (Rail Import)
-    section Port of Discharge
-        Vessel Arrival : Port of discharge : Terminal details : discharged timestamp
-        Vessel Birth : Port of discharge : Terminal details : Birth timestamp
-        Container Discharged : Port of discharge : Terminal details : discharged timestamp
-    section Container On Rail
-        Rail Loaded : Loaded location : loaded timestamp
-        Rail Departed : Departed location : departed timestamp
-        Rail Arrived : Arrived location : arrived timestamp
-        Rail Unloaded : Unloaded location : unloaded timestamp
-    section Final Destination (Rail Terminal)
-        Full Out : Full out port : Full out terminal : Full out timestamp
-        Empty Return : empty return timestamp
+### Webhook Notifications
 
-```
+Terminal49 provides webhook notifications to keep you updated on key milestones in a container's rail journey. These notifications allow you to integrate near real-time tracking data directly into your applications.
```

Comment:
Let's use 'transport events' in place of 'events' and 'milestones' to be consistent.

Reply:
Replaced all uses of "milestone" and some uses of "event" with "Transport Event".  Kept some as just "event" because changing them all made it read stilted and formal.